### DISCUSSION TITLE
Form Field - Hooks to intercept value before retrieve and before saving

### DISCUSSION THREAD

**Discussion by gthedev** [filamentphp/filament#10696] [Help]
I come from a Nova background and this could be achieved by using `retrieveUsing` and `fillUsing` callbacks on a Field, which means if there is a scenario that doesn't fit into a normal Field/Relation operation, you can completely customise on how the value for the field is being retrieved from a Model, and how it is being written to a Model (post validation). Is there an equivalent in Filament?

Current situation I have is a model with `HasMany` Relation, but the related model has values that are being driven by an Enum. So I want a `CheckboxList` with all of the Enum values, and on save the system will create & delete necessary `HasMany` relation records.

Thank you for your help!

**Comment by leandrocfe** [comment]
Welcome!
You can use `afterStateHydrated`
https://filamentphp.com/docs/3.x/forms/advanced#field-hydration

and `saveRelationshipsUsing`

```php
->saveRelationshipsUsing(function (array $state, CheckboxList $component, Model $record) {
    // TODO
})
```
